src/main.rs

```diff
@@ -61,6 +63,15 @@ fn main() -> io::Result<()> {
                 // Handle events and map to a Message
                 let mut current_msg = soko_game::handle_event(&mut model)?;
 
+                // When you win a level, move to the next level!
+                // XXX: This has to happen before the while loop below. Why?
```

Comment:
The loop Borrows current message I think so you cant use it again after the loop. You need to change the loop to a &ref or clone I think to fix.